### ISSUE TITLE
Making changes as we go clobbers semantic info

### ISSUE THREAD

**Issue by jschneidereit** [jschneidereit/MsTestToXunitConverter#13]
Basically the code in 6d75cf816fbcfa8548ecda3169ccb3992f95d4a9 was a really long and unpleasant way to figure out that almost everything in this project is wrong or bad. 

Instead of making changes as we determine that they need to be made, we need to queue the changes to be made after the analysis.

What I've found is that references to nodes in the code need to be reacquired before actually doing anything if a change is made.

So what needs to happen...I think... is that we need to store some kind of structure containing a hard position to a bit of code that can't be "lost" and a transform function. Then once the changes have been queued up, let them rip using `public static TRoot ReplaceNodes<TRoot, TNode>(this TRoot root, IEnumerable<TNode> nodes, Func<TNode, TNode, SyntaxNode> computeReplacementNode)` found in `Microsoft.CodeAnalysis.SyntaxNodeExtensions`.

**Comment by RikkiGibson** [comment]
Yeah, just come up with a list of nodes you want to do a particular kind of replacement on (e.g. MSTest's Assert.Equals to XUnit's Assert.Equals) and a function that does that kind of replacement. Come up with all the IEnumerables in advance based on semantic analysis and then ride a chain of .ReplaceNodes to victory.
